src/services/suggestionDiagnostics.ts

```diff
@@ -1,8 +1,28 @@
 /* @internal */
 namespace ts {
     export function computeSuggestionDiagnostics(sourceFile: SourceFile): Diagnostic[] {
-        return sourceFile.commonJsModuleIndicator
-            ? [createDiagnosticForNode(sourceFile.commonJsModuleIndicator, Diagnostics.File_is_a_CommonJS_module_it_may_be_converted_to_an_ES6_module)]
-            : emptyArray;
+        const diags: Diagnostic[] = [];
+
+        if (sourceFile.commonJsModuleIndicator) {
+            diags.push(createDiagnosticForNode(sourceFile.commonJsModuleIndicator, Diagnostics.File_is_a_CommonJS_module_it_may_be_converted_to_an_ES6_module));
+        }
+
+        function check(node: Node) {
+            switch (node.kind) {
+                case SyntaxKind.FunctionDeclaration:
+                case SyntaxKind.FunctionExpression:
+                    if (isInJavaScriptFile(node)) {

```

Comment:
since this is the only one we have right now, i would move this to the top of the `check` function. we can refactor it later as we add more checks.